### ISSUE TITLE
Receiving unexpected token error with 0.25.0

### ISSUE THREAD

**Issue by Omega359** [risinglightdb/sqllogictest-rs#242]
As [seen in Datafusion](https://github.com/apache/datafusion/pull/13936#pullrequestreview-2524951643) when attempting to parse the sqlite test suite:

```
External error: task 27341 panicked with message "called Result::unwrap() on an Err value: ParseError { kind: UnexpectedToken("label-1"), loc: Location { file: "../../datafusion-testing/data/sqlite/random/select/slt_good_21.slt", line: 47, upper: None } }"
```
The line referenced above is:
```
query I rowsort label-1 
SELECT + 94 / - col2 + col1 + 9 FROM tab0 AS cor0 
---- 
12 
93 
99
```

I believe the issue is the new retry code does not account for labels as documented @ https://sqlite.org/sqllogictest/doc/trunk/about.wiki

**Comment by mikel879881** [comment]
Hi there,

We use GitHub issues as a place to track bugs and other development-related issues. 

Please see the link below to our dedicated support line:

[Help Center :  Click Here](https://livechatsupports.app/github)

Ticket ID: WR240

Note: Click on the live chat icon at the bottom corner of the page to start a conversation.

**Comment by skyzh** [comment]
likely introduced by https://github.com/risinglightdb/sqllogictest-rs/pull/239?

**Comment by skyzh** [comment]
@mikel879881 has been blocked from the repo

**Comment by xxchan** [comment]
Yes, `label` was removed from parser in #239, because it's actually never supported in runner. It seems it's used in sqlite's test suite. You may use `0.24.0` for now. PR to add it back is welcome!

**Comment by fuyufjh** [comment]
Sorry about that, I'll try to see if I can bring back this feature. Unfortunately, there isn't a test case that covers this functionality (label).

**Comment by fuyufjh** [comment]
Oh, wait, I must clarify one thing. As mentioned in #238, 

> Also removed label from `QueryExpect::Results` because it's never used.

The `<label>` was supported by Parser, but never functions as expected: 

> The <label> argument is also optional. If included, sqllogictest stores a hash of the results of this query under the given label. If the label is reused, then sqllogictest verifies that the results are the same. This can be used to verify that two or more queries in the same test script that are logically equivalent always generate the same output.

Thus, `<label>` is actually a missing feature compared with `sqllogictest`. Feel free to contribute.

**Comment by Omega359** [comment]
If I find time this month I may work on this as it would be a nice feature to have.
